modules/gapi/include/opencv2/gapi/infer.hpp

```diff
@@ -398,32 +490,137 @@ struct Generic { };
 template<typename T = Generic> GInferOutputs
 infer(const std::string& tag, const GInferInputs& inputs)
 {
-    std::vector<GArg> input_args;
-    std::vector<std::string> input_names;
+    std::vector<cv::GArg> args;
+    std::vector<std::string> names;
+    cv::GKinds kinds;
 
-    const auto& blobs = inputs.getBlobs();
-    for (auto&& p : blobs)
-    {
-        input_names.push_back(p.first);
-        input_args.emplace_back(p.second);
-    }
-
-    GKinds kinds(blobs.size(), cv::detail::OpaqueKind::CV_MAT);
-    auto call = std::make_shared<cv::GCall>(GKernel{
-                GInferBase::id(),
-                tag,
-                GInferBase::getOutMeta,
-                {}, // outShape will be filled later
-                std::move(kinds),
-                {}, // outCtors will be filled later
-            });
+    unpackBlobs(inputs.getBlobs(), args, names, kinds);
 
-    call->setArgs(std::move(input_args));
-    call->params() = InOutInfo{input_names, {}};
+    auto call = makeCall<GInferBase>(tag,
```

Comment:
What's the reason to use `const &` and `&&` here?

Reply:
What do u mean ?